File: Design2_1.java
import java.util.Stack;

// Time Complexity : O(N)
// Space Complexity : O(N)
// Did this code successfully run on Leetcode : YES
// Any problem you faced while coding this : Took Hint to figure out the logic part


// Your code here along with comments explaining your approach

class MyQueue {

    /** Initialize your data structure here. */
    public MyQueue() {
        
    }
    
    Stack<Integer> stk1 = new Stack<>();
    Stack<Integer> stk2 = new Stack<>();
    /** Push element x to the back of queue. */
    public void push(int x) {
        if(stk1.empty()){
            stk1.push(x);
        }else{
            while(!stk1.empty()){
                stk2.push(stk1.pop());
            }
            
            stk1.push(x);
            
            while(!stk2.empty()){
                stk1.push(stk2.pop());
            }
        }
    }
    
    /** Removes the element from in front of queue and returns that element. */
    public int pop() {
        // if(!stk1.empty()){
            return stk1.pop();
        // }

    }
    
    /** Get the front element. */
    public int peek() {

            return stk1.peek();

    }
    
    /** Returns whether the queue is empty. */
    public boolean empty() {
        if(stk1.size()==0){
            return true;
        }
        
        return false;
    }
}

File: Design_2-1.py
class MyQueue:

    def __init__(self):
        self.S1=[]
        self.S2=[]   

    def push(self, x: int) -> None:
        self.S1.append(x)
        
    def pop(self) -> int:
        self.peek()
        return self.S2.pop()
    
    def peek(self) -> int:
        if not self.S2:
            while self.S1:
                self.S2.append(self.S1.pop())
        return self.S2[-1]
        
    def empty(self) -> bool:
        return not self.S1 and not self.S2
        


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()